zone/lua_client.cpp

```diff
@@ -19,3011 +18,484 @@
-}
-
-void Lua_Client::ResetCastbarCooldownBySlot(int slot) {
-	Lua_Safe_Call_Void();
-	self->ResetCastbarCooldownBySlot(slot);
-}
-
-void Lua_Client::ResetAllCastbarCooldowns() {
-	Lua_Safe_Call_Void();
-	self->ResetAllCastbarCooldowns();
-}
-
-void Lua_Client::ResetCastbarCooldownBySpellID(uint32 spell_id) {
-	Lua_Safe_Call_Void();
-	self->ResetCastbarCooldownBySpellID(spell_id);
-}
-
-void Lua_Client::UnscribeSpellBySpellID(uint16 spell_id) {
-	Lua_Safe_Call_Void();
-	self->UnscribeSpellBySpellID(spell_id);
-}
-
-void Lua_Client::UnscribeSpellBySpellID(uint16 spell_id, bool update_client) {
-	Lua_Safe_Call_Void();
-	self->UnscribeSpellBySpellID(spell_id, update_client);
-}
-
-int Lua_Client::GetEnvironmentDamageModifier() {
-	Lua_Safe_Call_Int();
-	return self->GetEnvironmentDamageModifier();
-}
-
-void Lua_Client::SetEnvironmentDamageModifier(int value) {
-	Lua_Safe_Call_Void();
-	self->SetEnvironmentDamageModifier(value);
-}
-
-bool Lua_Client::GetInvulnerableEnvironmentDamage() {
-	Lua_Safe_Call_Bool();
-	return self->GetInvulnerableEnvironmentDamage();
-}
-
-void Lua_Client::SetInvulnerableEnvironmentDamage(bool value) {
-	Lua_Safe_Call_Void();
-	self->SetInvulnerableEnvironmentDamage(value);
-}
-
-void Lua_Client::AddItem(luabind::object item_table) {
-	Lua_Safe_Call_Void();
-	if (luabind::type(item_table) != LUA_TTABLE) {
-		return;
-	}
-
-	uint32 item_id = luabind::object_cast<uint32>(item_table["item_id"]);
-	int16 charges = luabind::object_cast<uint32>(item_table["charges"]);
-	uint32 augment_one = luabind::type(item_table["augment_one"]) != LUA_TNIL ? luabind::object_cast<uint32>(item_table["augment_one"]) : 0;
-	uint32 augment_two = luabind::type(item_table["augment_two"]) != LUA_TNIL ? luabind::object_cast<uint32>(item_table["augment_two"]) : 0;
-	uint32 augment_three = luabind::type(item_table["augment_three"]) != LUA_TNIL ? luabind::object_cast<uint32>(item_table["augment_three"]) : 0;
-	uint32 augment_four = luabind::type(item_table["augment_four"]) != LUA_TNIL ? luabind::object_cast<uint32>(item_table["augment_four"]) : 0;
-	uint32 augment_five = luabind::type(item_table["augment_five"]) != LUA_TNIL ? luabind::object_cast<uint32>(item_table["augment_five"]) : 0;
-	uint32 augment_six = luabind::type(item_table["augment_six"]) != LUA_TNIL ? luabind::object_cast<uint32>(item_table["augment_six"]) : 0;
-	bool attuned = luabind::type(item_table["attuned"]) != LUA_TNIL ? luabind::object_cast<bool>(item_table["attuned"]) : false;
-	uint16 slot_id = luabind::type(item_table["slot_id"]) != LUA_TNIL ? luabind::object_cast<uint16>(item_table["slot_id"]) : EQ::invslot::slotCursor;
-
-	self->SummonItem(
-		item_id,
-		charges,
-		augment_one,
-		augment_two,
-		augment_three,
-		augment_four,
-		augment_five,
-		augment_six,
-		attuned,
-		slot_id
-	);
+	sv.new_enum("InventoryWhere", "Personal", invWherePersonal, "Bank", invWhereBank, "SharedBank",
+		    invWhereSharedBank, "Trading", invWhereTrading, "Cursor", invWhereCursor);
```

Comment:
Lost newlines on this one, also should it be in `lua_general_enums.cpp` with others?

Reply:
That would make sense, but it was here to start with and I didn't go back to move it.